geode-core/src/distributedTest/java/org/apache/geode/cache/client/internal/SSLDualServerNoClientAuthDUnitTest.java

```diff
@@ -0,0 +1,213 @@
+  private String hostName;
+
+  private static final String SERVER_1_KEYSTORE = "geodeserver1.keystore";
+  private static final String SERVER_1_TRUSTSTORE = "geodeserver1.truststore";
+
+  private static final String SERVER_2_KEYSTORE = "geodeserver2.keystore";
+  private static final String SERVER_2_TRUSTSTORE = "geodeserver2.truststore";
+
+
+  private static SSLDualServerNoClientAuthDUnitTest instance =
+      new SSLDualServerNoClientAuthDUnitTest();
+
+  @Before
+  public void setUp() {
+    disconnectAllFromDS();
+  }
+
+  @After
+  public void tearDown() {
+    VM serverVM = getVM(1);
+    VM server2VM = getVM(2);
+    VM locator = getVM(3);
+
+    locator.invoke(() -> closeLocatorTask());
+    server2VM.invoke(() -> closeCacheTask());
+    serverVM.invoke(() -> closeCacheTask());
+  }
+
+  @Test
+  public void testSSLServerWithNoAuth() {
+    VM serverVM = getVM(1);
+    VM server2VM = getVM(2);
+
+    VM locator = getVM(3);
+
+    Integer locatorPort = locator.invoke(() -> {
+      return setUpLocatorTask();
+    });
+    boolean cacheServerSslenabled = true;
+
+    serverVM.invoke(() -> setUpServerVMTask(locatorPort));
+    server2VM.invoke(() -> setUpServerVMTask(locatorPort));
+
+    server2VM.invoke(() -> doServerRegionTestTask());
+    serverVM.invoke(() -> doServerRegionTestTask());
+  }
+
+  private void createCache(Properties props) throws Exception {
+    cache = new CacheFactory(props).create();
+    if (cache == null) {
+      throw new Exception("CacheFactory.create() returned null ");
+    }
+  }
+
+  private Integer setUpLocator() throws Exception {
+    Properties gemFireProps = new Properties();
+
+    String cacheServerSslprotocols = "any";
+    String cacheServerSslciphers = "any";
+    boolean cacheServerSslRequireAuth = false;
+
+    System.setProperty("javax.net.debug", "all");
+
+    String keyStore =
+        createTempFileFromResource(SSLDualServerNoClientAuthDUnitTest.class, SERVER_1_KEYSTORE)
+            .getAbsolutePath();
+    String trustStore =
+        createTempFileFromResource(SSLDualServerNoClientAuthDUnitTest.class, SERVER_1_TRUSTSTORE)
+            .getAbsolutePath();
+    gemFireProps.setProperty(SSL_ENABLED_COMPONENTS, "cluster");
+    gemFireProps.setProperty(SSL_REQUIRE_AUTHENTICATION, "" + cacheServerSslRequireAuth);
+    gemFireProps.setProperty(SSL_KEYSTORE, "" + keyStore);
+    gemFireProps.setProperty(SSL_KEYSTORE_PASSWORD, "password");
+    gemFireProps.setProperty(SSL_TRUSTSTORE, "" + trustStore);
+    gemFireProps.setProperty(SSL_TRUSTSTORE_PASSWORD, "password");
+
+    StringWriter sw = new StringWriter();
+    PrintWriter writer = new PrintWriter(sw);
+    gemFireProps.list(writer);
+
+    Locator.startLocatorAndDS(0, new File(""), gemFireProps);
+
+    return Locator.getLocator().getPort();
+  }
+
+  private void setUpAndConnectToDistributedSystem(Integer locatorPort) throws Exception {
+    Properties gemFireProps = new Properties();
+
+    String cacheServerSslprotocols = "any";
```

Comment:
final

Reply:
According to Effective Java 3rd Edition, all local variables that are set only once are implicitly made `final` by the JVM without requiring use of the keyword. So it really just becomes noise (one more word on many lines). The only place where it's valuable is on class fields and constants.